umbraco/Umbraco-CMS
# Netcore: Remove UmbracoBackOfficeIdentity

The PR removes `UmbracoBackOfficeIdentity`, relying instead on `ClaimsIdentity`.

The primary approach is used here is making extension methods for what used to be in `UmbracoBackOfficeIdentity`, and replaces usages of `UmbracoBackOfficeIdentity` with `ClaimsIdentity`

### Details

In ClaimsPrincipalExtensions I retained GetUmbracoIdentity, but now it returns a ClaimsIdentity that is authenticated, and that has been verified to include all the required claims. I stuck with the UmbracoIdentity name since you could still say that a ClaimsIdentiy with the required Claims is considered an UmbracoBackOfficeIdentity.

---

Where we before instatiated an UmbracoBackOfficeIdentity we now instantiate a ClaimsIdentity and add all claims with the extension method AddRequiredClaims

---

UmbracoBackOfficeIdentity used to have a property for the default Issuer, this has also been moved to the extension class, this does lead to a `using ClaimsIdentityExtensions = Umbraco.Extensions.ClaimsIdentityExtensions;` which is less than nice, suggestions for what we can do instead would be appreciated.

---

I did have some issues with the RequiredBackOfficeClaimTypes enumerable (previously in UmbracoBackOfficeIdentity, now in ClaimsIdentityExtensions), because `StartContentNodeIdClaimType` and `StartMediaNodeIdClaimType` are listed as required, however these can be null if nothing is selected, for now I've just commented them out, and everything seems to work fine, but suggestions on what best to do here would also be appreciated.

---

I retained the UmbracoBackOfficeIdentity tests, but adjusted them to now be testing the extension methods instead


### Testing

Everything should be functionally the same, but some things you can test are
* Ensure that a site can still isntall proberly
* Try creating a new user, with different claims, start nodes in particular and ensure that they're enforced.
